src/torcontrol.cpp

```diff
@@ -285,17 +290,61 @@ static std::map<std::string,std::string> ParseTorReplyMapping(const std::string
             ++ptr; // skip opening '"'
             bool escape_next = false;
             while (ptr < s.size() && (escape_next || s[ptr] != '"')) {
```

Comment:
Is it illegal that a string will end with `\` ?
For example, say I want to represent `\` by  "\\",
I think this code would end up advancing ptr to s.size() on such an input and would return an empty string  in line 299